tests/next_tests/integration_tests/feature_tests/ffront_tests/test_concat_where.py

```diff
@@ -20,116 +22,360 @@
+    )()
+    air = cases.allocate(cartesian_case, testee, "air", domain=out.domain.slice_at[:, :, 1:])()
+
+    ref = np.concatenate((ground.asnumpy(), air.asnumpy()), axis=2)
+    cases.verify(cartesian_case, testee, ground, air, out=out, ref=ref)
+
+
+def test_concat_where_scalar_broadcast(cartesian_case):
+    @gtx.field_operator
+    def testee(a: np.int32, b: cases.IJKField, N: np.int32) -> cases.IJKField:
+        return concat_where(KDim < N - 1, a, b)
+
+    a = 3
+    b = cases.allocate(cartesian_case, testee, "b")()
+    out = cases.allocate(cartesian_case, testee, cases.RETURN)()
+
+    ref = np.concatenate(
+        (
+            np.full((*out.domain.shape[0:2], out.domain.shape[2] - 1), a),
+            b.asnumpy()[:, :, -1:],
+        ),
+        axis=2,
+    )
+    cases.verify(cartesian_case, testee, a, b, cartesian_case.default_sizes[KDim], out=out, ref=ref)
+
+
+def test_concat_where_scalar_broadcast_on_empty_branch(cartesian_case):
+    @gtx.field_operator
+    def testee(a: np.int32, b: cases.KField, N: np.int32) -> cases.KField:
+        return concat_where(KDim < N, a, b)
+
+    a = 3
+    b = cases.allocate(cartesian_case, testee, "b")()
+    out = cases.allocate(cartesian_case, testee, cases.RETURN, domain=b.domain.slice_at[1:])()
+
+    ref = b.asnumpy()[1:]
+    cases.verify(cartesian_case, testee, a, b, 1, out=out, ref=ref)
+
+
+def test_concat_where_single_level_broadcast(cartesian_case):
+    @gtx.field_operator
+    def testee(a: cases.KField, b: cases.IJKField) -> cases.IJKField:
+        return concat_where(KDim == 0, a, b)
+
+    out = cases.allocate(cartesian_case, testee, cases.RETURN)()
+    a = cases.allocate(
+        cartesian_case, testee, "a", domain=gtx.domain({KDim: out.domain.shape[2]})
+    )()
+    b = cases.allocate(cartesian_case, testee, "b", domain=out.domain.slice_at[:, :, 1:])()
+
+    ref = np.concatenate(
+        (
+            np.tile(a.asnumpy()[0], (*b.domain.shape[0:2], 1)),
+            b.asnumpy(),
+        ),
+        axis=2,
+    )
+    cases.verify(cartesian_case, testee, a, b, out=out, ref=ref)
+
+
+def test_concat_where_single_level_restricted_domain_broadcast(cartesian_case):
+    @gtx.field_operator
+    def testee(a: cases.KField, b: cases.IJKField) -> cases.IJKField:
+        return concat_where(KDim == 0, a, b)
+
+    out = cases.allocate(cartesian_case, testee, cases.RETURN)()
+    # note: this field is only defined on K: 0, 1, i.e., contains only a single value
+    a = cases.allocate(cartesian_case, testee, "a", domain=gtx.domain({KDim: (0, 1)}))()
+    b = cases.allocate(cartesian_case, testee, "b", domain=out.domain.slice_at[:, :, 1:])()
+
+    ref = np.concatenate(
+        (
+            np.tile(a.asnumpy()[0], (*b.domain.shape[0:2], 1)),
+            b.asnumpy(),
+        ),
+        axis=2,
+    )
+    cases.verify(cartesian_case, testee, a, b, out=out, ref=ref)
+
+
+def test_boundary_single_layer_3d_bc(cartesian_case):
```

Comment:
isn't it the same as the 2nd test `test_concat_where`

Reply:
it's actually slightly different than the 2nd and 3rd regarding overlapping. here we overlap, but have a single layer, in the 2nd we have full overlap and in 3rd we have no overlap.